chaos-days/blog/2023-12-18-Dynamically-scaling-brokers/index.md

```diff
@@ -0,0 +1,697 @@
+        },
+        {
+          "Id": 4,
+          "State": "ACTIVE",
+          "Priority": 2
+        },
+        {
+          "Id": 5,
+          "State": "ACTIVE",
+          "Priority": 3
+        }
+      ]
+    },
+    {
+      "Id": 5,
+      "State": "ACTIVE",
+      "Version": 8,
+      "LastUpdatedAt": "2023-12-18T15:30:16.949863252Z",
+      "Partitions": [
+        {
+          "Id": 4,
+          "State": "ACTIVE",
+          "Priority": 1
+        },
+        {
+          "Id": 5,
+          "State": "ACTIVE",
+          "Priority": 2
+        },
+        {
+          "Id": 6,
+          "State": "ACTIVE",
+          "Priority": 3
+        }
+      ]
+    },
+    {
+      "Id": 0,
+      "State": "ACTIVE",
+      "Version": 52,
+      "LastUpdatedAt": "2023-12-18T15:30:20.920293674Z",
+      "Partitions": [
+        {
+          "Id": 1,
+          "State": "ACTIVE",
+          "Priority": 3
+        },
+        {
+          "Id": 5,
+          "State": "ACTIVE",
+          "Priority": 1
+        },
+        {
+          "Id": 6,
+          "State": "ACTIVE",
+          "Priority": 2
+        }
+      ]
+    }
+  ],
+  "LastChange": {
+    "Id": 14,
+    "Status": "COMPLETED",
+    "StartedAt": "2023-12-18T15:12:57.790824149Z",
+    "CompletedAt": "2023-12-18T15:30:20.920657536Z"
+  },
+  "PendingChange": null
+}
+```
+
+All partitions are reported as healthy and leadership is balanced:
+
+```bash
+$ zbchaos topology
+Node      |Partition 1         |Partition 2         |Partition 3         |Partition 4         |Partition 5         |Partition 6
+0         |LEADER (HEALTHY)    |                    |                    |                    |FOLLOWER (HEALTHY)  |FOLLOWER (HEALTHY)
+1         |FOLLOWER (HEALTHY)  |LEADER (HEALTHY)    |                    |                    |                    |FOLLOWER (HEALTHY)
+2         |FOLLOWER (HEALTHY)  |FOLLOWER (HEALTHY)  |LEADER (HEALTHY)    |                    |                    |
+3         |                    |FOLLOWER (HEALTHY)  |FOLLOWER (HEALTHY)  |LEADER (HEALTHY)    |                    |
+4         |                    |                    |FOLLOWER (HEALTHY)  |FOLLOWER (HEALTHY)  |LEADER (HEALTHY)    |
+5         |                    |                    |                    |FOLLOWER (HEALTHY)  |FOLLOWER (HEALTHY)  |LEADER (HEALTHY)
+```
+
+The operation succeeded in about 17 minutes, longer than usual because of the restarts:
+![](scaleup-completed.png)
+
+
+## Scaling down should be resilient to broker restarts
+
+Exactly like scaling up, scaling down is also a high-level operation that consists of many steps that need to be carried out by all brokers in the cluster.
+Before a broker can leave, another broker first needs to join the replication group to ensure that we maintain a replication factor of 3 at all times.
+
+### Expected
+
+Even when brokers are restarting, the scale operation should eventually succeed with the expected cluster topology as result.
```

Comment:
🔧 similar to above, clarify the initial position and describe the experiment a bit more in detail you want to do.